test/builder/custom_mocks_test.dart

```diff
@@ -1843,6 +1843,32 @@ void main() {
     expect(mocksContent, contains('Iterable<X1> m1<X1>(X1 Function(X)? f)'));
     expect(mocksContent, contains('Iterable<X1?> m2<X1>(X1 Function(X)? f)'));
   });
+
+  test('We preserve nested generic bounded type arguments', () async {
```

Comment:
Thanks for the test! Could you please try adding test cases to cover the items in my comment above? Thanks!